Add pain.001.001.09 and pain.008.001.08

tobischo opened the pull request:
This pull request adds support for generating pain.001.001.09 and pain.008.001.08 files.

The relevant difference is surprisingly small.

For pain.001.001.09 the difference for the requested execution date is that it now supports date and date time for facilitating instant transfers eventually. In this pull request I skipped the changes to allow passing a time or date time value instead of the date. It should be a relatively simple addition, however it does require adapting the `validate_requested_date_after` function to a higher degree than I would want to add in this pull request.

I would appreciate any feedback on this

Commit:
Add pain.001.001.09 and pain.008.001.08

MerciMax commented:
Hello @tobischo -> What's the status of this PR? Should we merge it anytime soon? cc @ledermann

tobischo commented:
As far as I am concerned it is ready.
The changes from a version we installed from an internal gem server have been in use in production since about the time I created this PR.

MerciMax commented:
@tobischo  Addressing this then:

>  In this pull request I skipped the changes to allow passing a time or date time value instead of the date. It should be a relatively simple addition, however it does require adapting the validate_requested_date_after function to a higher degree than I would want to add in this pull request. 

does this bothered you in production for v9 sepa_xml ?

tobischo commented:
Not so far, but we also didn't need instant payment support.

For regular sepa credit transfers, date is sufficient.

For supporting sepa instant transfers, date time would be required.

That is why I highlighted that I skipped it. It needs to be clear that it is a limitation and for you to decide if you would want to merge the change as is or whether you'd require it to be added before merging.

tobischo committed:
Fix copy paste issue and use group[:account].bic

tobischo committed:
Update test matrix

Ruby 3.2 is only end of life since March 2026, so probably still used
Rails 6.1 is end of life since 2024

tobischo commented:
@MerciMax maybe worth looking at https://github.com/AdVitam/sepa_king/pull/1

The bug mentioned in there I also addressed here. We could introduce the newer formats directly as well or leave it for a separate change.
My use case were for embedding in container format, where those specific versions I added here are required (for the specific version of the container format).

Since tests started failing for ActiveModel 6.1 I updated the ruby and active model versions to cover some that are maybe only somewhat recently end of life: https://endoflife.date/rails, https://endoflife.date/ruby

Regarding sepa instant payments, I would suggest adding a `requested_time` field, which should be filled instead of the `requested_date` field to make the semantics clearer. Only one of the two should be set. I would also consider rather adding that in a separate change for ease of code review.
